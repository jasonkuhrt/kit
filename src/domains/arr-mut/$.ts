export * as ArrMut from './$$.ts'
