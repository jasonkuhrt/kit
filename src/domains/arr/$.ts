/**
 * Array utilities for working with readonly and mutable arrays.
 *
 * Provides functional utilities for array operations including mapping, filtering,
 * type guards, and conversions. Emphasizes immutable operations and type safety.
 *
 * @category Core Data Structures
 */
export * as Arr from './$$.js'
