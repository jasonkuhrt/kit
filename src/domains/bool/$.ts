export * as Bool from './$$.js'
