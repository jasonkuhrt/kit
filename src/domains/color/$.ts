// @ts-expect-error Duplicate identifier
export * as Color from './$$.js'

/**
 * Color utilities for terminal output.
 *
 * Provides comprehensive color support for CLI applications including:
 * - **140 CSS Level 4 named colors** (red, blue, etc.)
 * - **Multiple input formats**: hex, RGB, HSL, CSS functions, and objects
 * - **RGB (24-bit true color)** as canonical format
 * - **Effect Schema validation** with branded types
 * - **Type-level parser** for compile-time hints
 * - **Color conversions** between formats
 *
 * ## Supported Formats
 *
 * - Hex: `#FF5733` or `FF5733`
 * - RGB space-separated: `rgb 255 87 51`
 * - RGB CSS function: `rgb(255, 87, 51)`
 * - HSL space-separated: `hsl 120 100 50`
 * - HSL CSS function: `hsl(120, 100, 50)`
 * - Named colors: `red`, `blue`, `green`, etc. (140 colors)
 * - Object: `{ r: 255, g: 87, b: 51 }`
 *
 * ## Color Properties
 *
 * Colors can be applied to three targets:
 * - **foreground**: Text color
 * - **background**: Background color
 * - **underlineColor**: Underline color (when used with underline modifier)
 *
 * ## Usage
 *
 * @example
 * ```typescript
 * import { Color } from '@wollybeard/kit/color'
 *
 * // Parse various formats
 * const red = Color.fromString('red')
 * const blue = Color.fromString('#0000FF')
 * const green = Color.fromString('rgb 0 255 0')
 * const yellow = Color.fromString('hsl(60, 100, 50)')
 *
 * // Convert to hex
 * red.toHex()  // '#FF0000'
 *
 * // Use in schemas
 * const ConfigSchema = S.Struct({
 *   primaryColor: Color.String,
 *   secondaryColor: ColorInput
 * })
 * ```
 *
 * @category Color
 */
export namespace Color {}
