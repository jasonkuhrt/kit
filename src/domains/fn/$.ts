/**
 * Function utilities for functional programming patterns.
 *
 * Provides utilities for function composition, currying, partial application,
 * piping, and common functional patterns like identity and constant functions.
 * Includes function analysis tools and endomorphism utilities.
 *
 * @category Core Data Structures
 */
export * as Fn from './$$.js'
