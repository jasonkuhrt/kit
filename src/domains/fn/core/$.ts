export * as CoreFn from './$$.js'
