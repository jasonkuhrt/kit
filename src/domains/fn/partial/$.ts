export * as Partial from './$$.js'
