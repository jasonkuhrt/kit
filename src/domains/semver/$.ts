export * as Semver from './$$.js'
