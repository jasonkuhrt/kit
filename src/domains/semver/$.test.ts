import { S } from '#deps/effect'
import { Test } from '#test'
import { Semver } from './$.js'

const decodeSync = S.decodeSync(Semver.Semver)

Test
  .describe('decodeSync')
  .on(decodeSync)
  .casesInput(
    ['1.2.3'],
    ['1.2.3-beta.1+build.123'],
    ['invalid'],
  )
  .describeInputs('discriminated union', [
    ['1.2.3'],
    ['1.2.3+build.123'],
    ['1.2.3-beta'],
    ['1.2.3-beta+build.123'],
  ])
  .test()

Test
  .on(Semver.order)
  .cases(
    [[decodeSync('1.0.0'), decodeSync('2.0.0')], -1],
    [[decodeSync('2.0.0'), decodeSync('1.0.0')], 1],
    [[decodeSync('1.0.0'), decodeSync('1.0.0')], 0],
  )
  .test()

Test
  .on(Semver.equivalence)
  .cases(
    [[decodeSync('1.0.0'), decodeSync('1.0.0')], true],
    [[decodeSync('1.0.0'), decodeSync('2.0.0')], false],
  )
  .test()

Test
  .on(Semver.is)
  .cases(
    [[decodeSync('1.2.3')], true],
    [[{ major: 1, minor: 2, patch: 3 }], false],
    [['1.2.3'], false],
  )
  .test()

Test
  .on(Semver.increment)
  .casesInput(
    [decodeSync('1.2.3'), 'major'],
    [decodeSync('1.2.3'), 'minor'],
    [decodeSync('1.2.3'), 'patch'],
  )
  .test()

Test
  .on(Semver.min)
  .casesInput(
    [decodeSync('1.0.0'), decodeSync('2.0.0')],
    [decodeSync('2.0.0'), decodeSync('1.0.0')],
  )
  .test()

Test
  .on(Semver.max)
  .casesInput(
    [decodeSync('1.0.0'), decodeSync('2.0.0')],
    [decodeSync('2.0.0'), decodeSync('1.0.0')],
  )
  .test()

Test
  .on(Semver.make)
  .casesInput(
    [1, 2, 3],
    [1, 2, 3, 'beta.1'],
    [1, 2, 3, 'beta.1', 'build.123'],
    [1, 2, 3, undefined, 'build.123'],
  )
  .test()

Test
  .on(Semver.satisfies)
  .onSetup(() => ({
    v: decodeSync('1.2.3'),
  }))
  .casesInput(
    // todo: this should work:
    // ({ v }) => [v, '>=1.0.0'],
    [decodeSync('1.2.3'), '>=1.0.0'],
    [decodeSync('1.2.3'), '^1.0.0'],
    [decodeSync('1.2.3'), '~1.2.0'],
    [decodeSync('1.2.3'), '2.x'],
    [decodeSync('1.2.3'), 'invalid'],
  )
  .test()
