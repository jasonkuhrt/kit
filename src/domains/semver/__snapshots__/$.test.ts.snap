// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`decodeSync > fn('1.2.3') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`decodeSync > fn('1.2.3-beta.1+build.123') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3-beta.1+build.123
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
PreRelease {
  major: 1,
  minor: 2,
  patch: 3,
  prerelease: [ 'beta', 1 ],
  build: [ 'build', '123' ],
  version: Version {
    raw: '1.2.3-beta.1+build.123',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: [ 'beta', 1 ],
    build: [ 'build', '123' ]
  },
  _tag: 'SemverPreRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`decodeSync > fn('invalid') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
invalid
╠══════════════════════════════════════════════════╣ THEN THROWS PARSEERROR
ParseError: (string <-> Semver)
└─ Transformation process failure
   └─ Invalid semver: SyntaxError: invalid version: no Major.minor.patch tuple present
╚══════════════════════════════════════════════════╝"
`;

exports[`discriminated union > decodeSync > fn('1.2.3') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`discriminated union > decodeSync > fn('1.2.3+build.123') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3+build.123
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: [ 'build', '123' ],
  version: Version {
    raw: '1.2.3+build.123',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: [ 'build', '123' ]
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`discriminated union > decodeSync > fn('1.2.3-beta') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3-beta
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
PreRelease {
  major: 1,
  minor: 2,
  patch: 3,
  prerelease: [ 'beta' ],
  build: undefined,
  version: Version {
    raw: '1.2.3-beta',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: [ 'beta' ],
    build: undefined
  },
  _tag: 'SemverPreRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`discriminated union > decodeSync > fn('1.2.3-beta+build.123') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3-beta+build.123
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
PreRelease {
  major: 1,
  minor: 2,
  patch: 3,
  prerelease: [ 'beta' ],
  build: [ 'build', '123' ],
  version: Version {
    raw: '1.2.3-beta+build.123',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: [ 'beta' ],
    build: [ 'build', '123' ]
  },
  _tag: 'SemverPreRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn('1.2.3') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn('1.2.3') 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn('1.2.3+build.123') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3+build.123
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: [ 'build', '123' ],
  version: Version {
    raw: '1.2.3+build.123',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: [ 'build', '123' ]
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn('1.2.3-beta') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3-beta
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
PreRelease {
  major: 1,
  minor: 2,
  patch: 3,
  prerelease: [ 'beta' ],
  build: undefined,
  version: Version {
    raw: '1.2.3-beta',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: [ 'beta' ],
    build: undefined
  },
  _tag: 'SemverPreRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn('1.2.3-beta+build.123') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3-beta+build.123
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
PreRelease {
  major: 1,
  minor: 2,
  patch: 3,
  prerelease: [ 'beta' ],
  build: [ 'build', '123' ],
  version: Version {
    raw: '1.2.3-beta+build.123',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: [ 'beta' ],
    build: [ 'build', '123' ]
  },
  _tag: 'SemverPreRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn('1.2.3-beta.1+build.123') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.2.3-beta.1+build.123
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
PreRelease {
  major: 1,
  minor: 2,
  patch: 3,
  prerelease: [ 'beta', 1 ],
  build: [ 'build', '123' ],
  version: Version {
    raw: '1.2.3-beta.1+build.123',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: [ 'beta', 1 ],
    build: [ 'build', '123' ]
  },
  _tag: 'SemverPreRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn('invalid') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
invalid
╠══════════════════════════════════════════════════╣ THEN THROWS PARSEERROR
ParseError: (string <-> Semver)
└─ Transformation process failure
   └─ Invalid semver: SyntaxError: invalid version: no Major.minor.patch tuple present
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 1, minor: 0, patch: 0, build: undefined, version: Versi..., OfficialRelease { major: 2, minor: 0, patch: 0, build: undefined, version: Versi...) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '1.0.0',
    major: 1,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
OfficialRelease {
  major: 2,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '2.0.0',
    major: 2,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '1.0.0',
    major: 1,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 1, minor: 0, patch: 0, build: undefined, version: Versi..., OfficialRelease { major: 2, minor: 0, patch: 0, build: undefined, version: Versi...) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '1.0.0',
    major: 1,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
OfficialRelease {
  major: 2,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '2.0.0',
    major: 2,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 2,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '2.0.0',
    major: 2,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 1, minor: 2, patch: 3, build: undefined, version: Versi..., '>=1.0.0') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
>=1.0.0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 1, minor: 2, patch: 3, build: undefined, version: Versi..., '^1.0.0') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
^1.0.0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 1, minor: 2, patch: 3, build: undefined, version: Versi..., '~1.2.0') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
~1.2.0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 1, minor: 2, patch: 3, build: undefined, version: Versi..., '2.x') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
2.x
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 1, minor: 2, patch: 3, build: undefined, version: Versi..., 'invalid') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
invalid
╠══════════════════════════════════════════════════╣ THEN THROWS SYNTAXERROR
SyntaxError: invalid comparator: 'invalid' major must be numeric or 'x', got: 'invalid'
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 2, minor: 0, patch: 0, build: undefined, version: Versi..., OfficialRelease { major: 1, minor: 0, patch: 0, build: undefined, version: Versi...) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 2,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '2.0.0',
    major: 2,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
OfficialRelease {
  major: 1,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '1.0.0',
    major: 1,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '1.0.0',
    major: 1,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`fn(OfficialRelease { major: 2, minor: 0, patch: 0, build: undefined, version: Versi..., OfficialRelease { major: 1, minor: 0, patch: 0, build: undefined, version: Versi...) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 2,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '2.0.0',
    major: 2,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
OfficialRelease {
  major: 1,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '1.0.0',
    major: 1,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 2,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '2.0.0',
    major: 2,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`major > increment(OfficialRelease { major: 1, minor: 2, patch: 3, build: undefined, version: Versi..., 'major') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
major
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 2,
  minor: 0,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '2.0.0',
    major: 2,
    minor: 0,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`make(1, 2, 3) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
────────────────────────────────────────────────────
2
────────────────────────────────────────────────────
3
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`make(1, 2, 3, 'beta.1') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
────────────────────────────────────────────────────
2
────────────────────────────────────────────────────
3
────────────────────────────────────────────────────
beta.1
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
PreRelease {
  major: 1,
  minor: 2,
  patch: 3,
  prerelease: [ 'beta', 1 ],
  build: undefined,
  version: Version {
    raw: '1.2.3-beta.1',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: [ 'beta', 1 ],
    build: undefined
  },
  _tag: 'SemverPreRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`make(1, 2, 3, 'beta.1', 'build.123') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
────────────────────────────────────────────────────
2
────────────────────────────────────────────────────
3
────────────────────────────────────────────────────
beta.1
────────────────────────────────────────────────────
build.123
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
PreRelease {
  major: 1,
  minor: 2,
  patch: 3,
  prerelease: [ 'beta', 1 ],
  build: [ 'build', '123' ],
  version: Version {
    raw: '1.2.3-beta.1+build.123',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: [ 'beta', 1 ],
    build: [ 'build', '123' ]
  },
  _tag: 'SemverPreRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`make(1, 2, 3, undefined, 'build.123') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
────────────────────────────────────────────────────
2
────────────────────────────────────────────────────
3
────────────────────────────────────────────────────
undefined
────────────────────────────────────────────────────
build.123
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: [ 'build', '123' ],
  version: Version {
    raw: '1.2.3+build.123',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: [ 'build', '123' ]
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`minor > increment(OfficialRelease { major: 1, minor: 2, patch: 3, build: undefined, version: Versi..., 'minor') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
minor
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 3,
  patch: 0,
  build: undefined,
  version: Version {
    raw: '1.3.0',
    major: 1,
    minor: 3,
    patch: 0,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;

exports[`patch > increment(OfficialRelease { major: 1, minor: 2, patch: 3, build: undefined, version: Versi..., 'patch') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 3,
  build: undefined,
  version: Version {
    raw: '1.2.3',
    major: 1,
    minor: 2,
    patch: 3,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
────────────────────────────────────────────────────
patch
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
OfficialRelease {
  major: 1,
  minor: 2,
  patch: 4,
  build: undefined,
  version: Version {
    raw: '1.2.4',
    major: 1,
    minor: 2,
    patch: 4,
    prerelease: undefined,
    build: undefined
  },
  _tag: 'SemverOfficialRelease'
}
╚══════════════════════════════════════════════════╝"
`;
